bundle/src/main/java/com/adobe/acs/commons/workflow/synthetic/impl/granite/SyntheticWorkflowSession.java

```diff
@@ -163,155 +149,32 @@ public final WorkflowModel getModel(final String modelId) throws WorkflowExcepti
-    }
-
-    @Override
-    public final List<Route> getRoutes(final WorkItem workItem, final boolean b) throws WorkflowException {
+    public final List<Route> getRoutes() throws WorkflowException {
         log.debug("Synthetic Workflow does not support routes; Defaults to a single Synthetic Route");
         return this.routes;
     }
 
-    @Override
-    public final List<Route> getBackRoutes(final WorkItem workItem, final boolean b) throws WorkflowException {
+    public final List<Route> getBackRoutes() throws WorkflowException {
         log.debug("Synthetic Workflow does not back support routes; Defaults to a single Synthetic Route");
         return this.backRoutes;
     }
 
-    @Override
     public final WorkflowData newWorkflowData(final String payloadType, final Object payload) {
         return new SyntheticWorkflowData(payloadType, payload);
     }
 
-    @Override
-    public Iterator<Participant> getDelegates(WorkItem workItem) throws WorkflowException {
-        throw new UnsupportedOperationException(UNSUPPORTED_OPERATION_MESSAGE);
-    }
-
-    @Override
-    public void delegateWorkItem(WorkItem workItem, Participant participant) throws WorkflowException, AccessControlException {
-        throw new UnsupportedOperationException(UNSUPPORTED_OPERATION_MESSAGE);
-    }
-
-    @Override
-    public final List<HistoryItem> getHistory(final Workflow workflow) throws WorkflowException {
-        throw new UnsupportedOperationException(UNSUPPORTED_OPERATION_MESSAGE);
-    }
-
-    @Override
     public final void updateWorkflowData(final Workflow workflow, final WorkflowData workflowData) {
         if (workflow instanceof com.adobe.acs.commons.workflow.synthetic.impl.granite.SyntheticWorkflow) {
             final SyntheticWorkflow syntheticWorkflow = (SyntheticWorkflow) workflow;
             syntheticWorkflow.setWorkflowData(workflowData);
         } else {
-            throw new UnsupportedOperationException(UNSUPPORTED_OPERATION_MESSAGE);
+            throw new UnsupportedOperationException(UNSUPPORTED_OPERATION_MESSAGE + "updateWorkflowData that does not contain a granite SyntheticWorkflow");
         }
     }
 
-    @Override
-    public final void logout() {
-        throw new UnsupportedOperationException(UNSUPPORTED_OPERATION_MESSAGE);
-    }
-
-    @Override
-    public <AdapterType> AdapterType adaptTo(Class<AdapterType> type) {
-        if (ResourceResolver.class == type) {
-            if (this.session != null) {
-                try {
-                    return (AdapterType) workflowService.getResourceResolver(session);
-                } catch (LoginException e) {
-                    log.error("Failed to adapt Synthetic Granite WorkflowSession to ResourceResolver", e);
-                }
-            }
-        } else if (Session.class == type) {
-            return (AdapterType) this.session;
+    public Object adaptTo(Class<?> type) {
```

Comment:
Any reason why this is not adaptable to a ResourceResolver anymore?

Reply:
Oh, I missed that somehow. I'll add it back in!